spec/models/claim_update_spec.rb

```diff
@@ -93,21 +112,79 @@
     end
 
     context "when the update changes employment_status to :different_school" do
-      let(:claim) { create(:claim, eligibility: build(:student_loans_eligibility, claim_school: schools(:penistone_grammar_school), employment_status: :claim_school, current_school: schools(:penistone_grammar_school))) }
+      let(:eligibility) do
+        build(
+          :student_loans_eligibility,
+          employment_status: :claim_school,
+          current_school: schools(:penistone_grammar_school),
+          employments: [employment]
+        )
+      end
+
+      let(:employment) do
+        build(
+          :student_loans_employment,
+          school: schools(:penistone_grammar_school)
+        )
+      end
+
+      let(:claim) do
+        create(
+          :claim,
+          eligibility: eligibility
+        )
+      end
+
       let(:context) { "still-teaching" }
-      let(:params) { {eligibility_attributes: {employment_status: "different_school"}} }
 
-      it "resets the inferrred current_school to nil" do
+      let(:params) do
+        {
+          eligibility_attributes: {
+            employment_status: "different_school",
+          },
+        }
+      end
+
+      it "resets the inferred current_school to nil" do
         expect(claim_update.perform).to be_truthy
         expect(claim.eligibility.reload.employment_status).to eq "different_school"
         expect(claim.eligibility.current_school).to be_nil
       end
     end
 
     context "when the update does not actually change the employment_status" do
-      let(:claim) { create(:claim, eligibility: build(:student_loans_eligibility, claim_school: schools(:penistone_grammar_school), employment_status: :different_school, current_school: schools(:hampstead_school))) }
+      let(:eligibility) do
+        build(
+          :student_loans_eligibility,
+          employment_status: :different_school,
+          current_school: schools(:hampstead_school),
+          employments: [employment]
+        )
+      end
+
+      let(:employment) do
+        build(
+          :student_loans_employment,
+          school: schools(:penistone_grammar_school)
+        )
+      end
+
+      let(:claim) do
+        create(
+          :claim,
+          eligibility: eligibility
+        )
+      end
```

Comment:
This can be further simplified by inlining the `employment` and `eligibility` builds in `claim`.